core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -148,11 +148,20 @@ private[spark] class AppStatusStore(
     // cheaper for disk stores (avoids deserialization).
     val count = {
       Utils.tryWithResource(
-        store.view(classOf[TaskDataWrapper])
-          .parent(stageKey)
-          .index(TaskIndexNames.EXEC_RUN_TIME)
-          .first(0L)
-          .closeableIterator()
+        if (store.isInstanceOf[LevelDB]) {
```

Comment:
Does this code path need to be different for disk vs memory? this part seemed like it could work efficiently either way.

Reply:
Yes. Now, for diskStore case, it finds total tasks count and inMemory case only successful tasks count.

This 'count' is used to find quantileIndices for all the tasks metrics.
https://github.com/apache/spark/blob/676bbb2446af1f281b8f76a5428b7ba75b7588b3/core/src/main/scala/org/apache/spark/status/AppStatusStore.scala#L222